### PR TITLE
No Erase

### PR DESCRIPTION
**Pull request by tsao-chi**
leftovers: 
+ abstract SubLevelDef.isProp() ?

### CODE REVIEW

File: base/src/main/java/org/aya/core/def/FieldDef.java [hunk 1]
```diff
@@ -33,4 +34,13 @@ public FieldDef(
...
   }
+
+  private @NotNull SortTerm structResult() {
+    return structRef.concrete != null ? structRef.concrete.ulift : structRef.core.result;
+  }
+
+
+  public boolean inProp() {
+    return structResult().isProp();
+  }
```
Comment (ice1000) [inline base/src/main/java/org/aya/core/def/FieldDef.java:45]: Is it possible to turn this into SubLevelDef

File: base/src/main/java/org/aya/tyck/StmtTycker.java [hunk 2]
```diff
@@ -243,7 +243,7 @@ public void tyckHeader(@NotNull Decl decl, @NotNull ExprTycker tycker) {
           // No patterns, leave it blank
           ctor.yetTyckedPat = ImmutableSeq.empty();
         }
-        var ctorSort = dataConcrete.ulift.kind() == SortKind.Prop ? SortTerm.Type0 : dataConcrete.ulift;
+        var ctorSort = dataConcrete.ulift.isProp() ? SortTerm.Type0 : dataConcrete.ulift;
```
Comment (ice1000) [inline base/src/main/java/org/aya/tyck/StmtTycker.java:246]: Can they have higher levels?

File: base/src/main/java/org/aya/tyck/ExprTycker.java [hunk 3]
```diff
@@ -58,7 +59,7 @@
  * Do <em>not</em> use multiple instances in the tycking of one {@link Decl.TopLevel}
  * and do <em>not</em> reuse instances of this class in the tycking of multiple {@link Decl.TopLevel}s.
  */
-public final class ExprTycker extends Tycker {
+public final class ExprTycker extends Tycker implements Cloneable {
```
Comment (ice1000) [inline base/src/main/java/org/aya/tyck/ExprTycker.java:62]: Why

File: base/src/main/java/org/aya/core/def/FieldDef.java (hunk 1 above)
Comment (tsao-chi) [reply base/src/main/java/org/aya/core/def/FieldDef.java:45]: SubLevel doesn’t have structRef

File: base/src/main/java/org/aya/tyck/ExprTycker.java (hunk 3 above)
Comment (tsao-chi) [reply base/src/main/java/org/aya/tyck/ExprTycker.java:62]: Oh

Comment (tsao-chi) [reply base/src/main/java/org/aya/tyck/ExprTycker.java]: fixed

File: base/src/main/java/org/aya/tyck/StmtTycker.java (hunk 2 above)
Comment (tsao-chi) [reply base/src/main/java/org/aya/tyck/StmtTycker.java]: How about 44154？ （bushi

Comment (ice1000) [reply base/src/main/java/org/aya/tyck/StmtTycker.java]: I mean is it theoretically sound?

File: base/src/main/java/org/aya/core/def/FieldDef.java (hunk 1 above)
Comment (ice1000) [reply base/src/main/java/org/aya/core/def/FieldDef.java:44]: I believe there will be a way to share some code

File: base/src/main/java/org/aya/core/term/SortTerm.java [hunk 4]
```diff
@@ -16,6 +16,7 @@ public SortTerm(@NotNull SortKind kind, int lift) {
   }
 
   public static SortTerm Type0 = new SortTerm(SortKind.Type, 0);
+  public static SortTerm Type44154 = new SortTerm(SortKind.Type, 44154);
```
Comment (ice1000) [inline base/src/main/java/org/aya/core/term/SortTerm.java:19]: Why?

**Review by ice1000 — CHANGES_REQUESTED** [review]

File: base/src/main/java/org/aya/core/term/SortTerm.java [hunk 5]
```diff
@@ -16,6 +16,7 @@ public SortTerm(@NotNull SortKind kind, int lift) {
...
   public static SortTerm Set0 = new SortTerm(SortKind.Set, 0);
   public static SortTerm Set1 = new SortTerm(SortKind.Set, 1);
```
Comment (imkiva) [inline base/src/main/java/org/aya/core/term/SortTerm.java:21]: Why `44154`?

File: base/src/main/java/org/aya/core/def/FieldDef.java (hunk 1 above)
Comment (tsao-chi) [reply base/src/main/java/org/aya/core/def/FieldDef.java:44]: It might not be trivial

**Comment by tsao-chi** [comment]
leftovers: 
+ more user friendly error message
+ Prop check for MetaTerm?

**Comment by ice1000** [comment]
Yes you need Prop check for MetaTerm

**Comment by tsao-chi** [comment]
> Yes you need Prop check for MetaTerm

implemented in the latest commit

**Review by ice1000 — APPROVED** [review]

**Pull request merged**